### PR TITLE
Add LIST_ENVIRONMENT_DATASETS permission for listing shared datasets and cleanup unused code

### PR DESCRIPTION
**Pull request by dlpzx**
### Feature or Bugfix
- Bugfix

### Detail
Added permission check on the list datasets API calls from the S3 shares module. Ensuring that only environment members can see environment shared datasets.

++ remove some unused code

### Security
Please answer the questions below briefly where applicable, or write `N/A`. Based on
[OWASP 10](https://owasp.org/Top10/en/).

- Does this PR introduce or modify any input fields or queries - this includes
fetching data from storage outside the application (e.g. a database, an S3 bucket)?
  - Is the input sanitized?
  - What precautions are you taking before deserializing the data you consume?
  - Is injection prevented by parametrizing queries?
  - Have you ensured no `eval` or similar functions are used?
- Does this PR introduce any functionality or component that requires authorization?
  - How have you ensured it respects the existing AuthN/AuthZ mechanisms?
  - Are you logging failed auth attempts?
- Are you using or adding any cryptographic features?
  - Do you use a standard proven implementations?
  - Are the used keys controlled by the customer? Where are they stored?
- Are you introducing any new policies/roles/users?
  - Have you used the least-privilege principle? How?

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: backend/dataall/modules/s3_datasets_shares/services/s3_share_service.py [hunk 1]
```diff
@@ -259,11 +261,17 @@ def get_s3_consumption_data(uri):
             }
 
     @staticmethod
-    def list_shared_databases_tables_with_env_group(environmentUri: str, groupUri: str):
+    @ResourcePolicyService.has_resource_permission(LIST_ENVIRONMENT_DATASETS)
+    def list_shared_databases_tables_with_env_group(uri: str, group_uri: str):
         context = get_context()
+        if group_uri not in context.groups:
```
Comment (SofiaSazonova) [inline backend/dataall/modules/s3_datasets_shares/services/s3_share_service.py:267]: This condition (or the similar ones) is widely used.
May be we can make a decorator out of it?

Comment (SofiaSazonova) [reply backend/dataall/modules/s3_datasets_shares/services/s3_share_service.py:267]: E.g. 
image

**Review by SofiaSazonova — APPROVED** [review]
Minor thing: more about refactoring

File: backend/dataall/modules/s3_datasets/services/dataset_profiling_service.py [hunk 2]
```diff
@@ -67,13 +67,6 @@ def resolve_profiling_run_status(run_uri):
             session.add(task)
         Worker.queue(engine=context.db_engine, task_ids=[task.taskUri])
 
-    @staticmethod
-    @ResourcePolicyService.has_resource_permission(GET_DATASET)
-    @is_feature_enabled('modules.s3_datasets.features.metrics_data')
-    def list_profiling_runs(uri):
-        with get_context().db_engine.scoped_session() as session:
-            return DatasetProfilingRepository.list_profiling_runs(session, uri)
-
```
Comment (noah-paige) [inline backend/dataall/modules/s3_datasets/services/dataset_profiling_service.py:76]: if removing here - should we not remove `def list_profiling_runs(session, dataset_uri)` from `dataall/backend/dataall/modules/s3_datasets/db/dataset_profiling_repositories.py` ?

File: backend/dataall/modules/s3_datasets/services/dataset_table_service.py [hunk 3]
```diff
@@ -39,11 +39,6 @@ def _get_dataset_uri(session, table_uri):
         table = DatasetTableRepository.get_dataset_table_by_uri(session, table_uri)
         return table.datasetUri
 
-    @staticmethod
-    def get_table(uri: str):
-        with get_context().db_engine.scoped_session() as session:
-            return DatasetTableRepository.get_dataset_table_by_uri(session, uri)
-
```
Comment (noah-paige) [inline backend/dataall/modules/s3_datasets/services/dataset_table_service.py:46]: why this?

File: backend/dataall/modules/s3_datasets_shares/services/s3_share_service.py [hunk 4]
```diff
@@ -259,11 +261,17 @@ def get_s3_consumption_data(uri):
...
+            raise exceptions.UnauthorizedOperation(
+                action='LIST_ENVIRONMENT_GROUP_DATASETS',
+                message=f'User: {context.username} is not a member of the team {group_uri}',
```
Comment (noah-paige) [inline backend/dataall/modules/s3_datasets_shares/services/s3_share_service.py:270]: nit: this is 1 example where we may not want to use input in output error message in the case of bad inputs being used to reflect back

File: backend/dataall/modules/s3_datasets/services/dataset_profiling_service.py (hunk 2 above)
Comment (dlpzx) [reply backend/dataall/modules/s3_datasets/services/dataset_profiling_service.py:76]: Done, cleaned up!

**Review by petrkalos — APPROVED** [review]

**Pull request merged**